File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    max_width = 0
    max_height = 0

    for w, h in sizes:
        width, height = max(w, h), min(w, h)
        max_width = max(max_width, width)
        max_height = max(max_height, height)

    return max_width * max_height

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.js
function solution(sizes) {
    // 먼저 지갑들 마다의 긴변, 짧은 변을 맞춘 다음에 비교 
    let width = 0;
    let height = 0;
    
    for(const [w, h] of sizes) {
        const long = Math.max(w, h);
        const short = Math.min(w, h);
        
        width = Math.max(width, long);
        height = Math.max(height, short);
    }
    
    return width * height;
}